src/kubernetes.cc

```diff
@@ -811,22 +813,21 @@ void KubernetesReader::WatchMaster(
     }
     try {
       if (verbose) {
-        LOG(INFO) << "Locking completion mutex";
+        LOG(INFO) << "Resetting and running service";
       }
-      // A notification for watch completion.
-      std::mutex completion_mutex;
-      std::unique_lock<std::mutex> watch_completion(completion_mutex);
+      service->reset();
+      auto lifetime_thread = std::thread([&service]() { service->run(); });
```

Comment:
Shared pointers are passed by copy. Plus, this is too verbose. So you could just write:
```c++
std::thread lifetime_thread([=]() { service->run(); });
```
But see below.

Reply:
Used the version below.